src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -396,6 +396,7 @@ protected virtual void Dispose(bool disposing)
         public virtual void RestartHost()
         {
             _restartHostEvent.Set();
+            State = ScriptHostState.Default;
```

Comment:
Why do we reset to Default here, instead of in the RunAndBlock loop where the rest of the state is managed?

Reply:
The reason was because in StandbyManager.Warmup, when I call RestartHost I immediately enter a DelayUntilHostReady loop. Without this change, that loop would immediately return because the host state hadn't yet reset (host hadn't started restart).

I didn't like doing this either, and vacillated on it a bit. I'll see if I can improve this.

Reply:
I'm just going to add a detailed comment on this and leave as is for now